ocs_ci/utility/rosa.py

```diff
@@ -230,6 +237,16 @@ def install_odf_addon(cluster):
         public_key_only = remove_header_footer_from_key(public_key)
         cmd += f' --onboarding-validation-key "{public_key_only}"'
 
+    if cluster_type.lower() == "consumer" and config.ENV_DATA.get("provider_name", ""):
```

Comment:
You already got the provider name in line 215. you can replace `config.ENV_DATA.get("provider_name", "")` with the var  `provider_name`